react/README.md

```diff
@@ -297,6 +298,74 @@
   ))}
   ```
 
+## Proptypes
+  - Always use `PropTypes.shape` to define object proptypes. Avoid `PropTypes.object`.
+    ```
+      UserDisplay.propTypes = {
+        user: PropTypes.shape({
+          name: PropTypes.string,
+          email: PropTypes.string,
+          birthday: PropTypes.instanceOf(Date),
+        }),
+      };
+    ```
+  - When a component's proptypes can be used in multiple components, export it as a `const` from the component file.
+    ```
+      export const statusPropTypes = {
+        active: PropTypes.bool.isRequired,
+        message: PropTypes.string.isRequired,
+      };
+      StatusMessage.propTypes = statusPropTypes;
+      export default StatusMessage;
+    ```
+
+    Usage:
+    ```
+      import StatusMessage, { statusPropTypes } from './status-message';
+
+      class AppContainer { };
+      AppContainer.propTypes = {
+        status: statusPropTypes,
+        data: PropTypes.string.isRequired,
+      };
+    ```
+  - When a subset of a component's proptypes can be used in multiple components, export it as a `const` from the component file.
+
+    ```
+      export const partialFilterPropTypes = {
+        searchString: PropTypes.string,
+        onFilter: PropTypes.func,
+      };
+      FilterData.propTypes = {
+        ...partialFilterPropTypes,
+        isFilterActive: PropTypes.bool.isRequired,
+      };
+    ```
+  - When a set of proptypes are not associated with a single component, but can be used across multiple components, define them in a `constants/proptypes.js` file. Example:
```

Comment:
is this specific to `shape`?  

if not, it's not entirely clear to me how to differentiate between the previous rule and this rule (when do we export the proptypes from a component, and when do we export it from `proptypes.js`?)